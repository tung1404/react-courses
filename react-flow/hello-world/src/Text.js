/* @flow */
import React from "react";
import "./Text.css";

type Props = { ten: string, removeGreeting: (name: string) => void };
type State = { name: string };

class Text extends React.Component<Props, State> {
  props: Props;
  state: State;

  constructor(props: Props) {
    super(props);
    this.state = { name: props.ten };
    this.removeGreeting = this.removeGreeting.bind(this);
  }

  removeGreeting = function() {
    this.props.removeGreeting(this.props.ten);
  };

  render() {
    return (
      <div className="Text">
        Hi, {this.state.name}!
        <br />
        <button onClick={this.removeGreeting}>Remove Me!</button>
      </div>
    );
  }
}

export default Text;
