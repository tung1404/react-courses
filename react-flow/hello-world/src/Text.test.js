import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Text from "./Text";

describe(Text, () => {
  // Add the rest of our tests here later!
  it("renders without crashing", () => {
    const name = "Person";
    const mockRemoveGreeting = jest.fn();
    const component = shallow(
      <Text ten={name} removeGreeting={mockRemoveGreeting} />
    );
    it("modifies the greeting when frenchify button is clicked", () => {
      component.find("button.frenchify").simulate("click");
      expect(component.text()).toContain("Bonjour");
    });
  });

  it("renders and matches our snapshot", () => {
    const component = renderer.create(<Text ten="Person" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
