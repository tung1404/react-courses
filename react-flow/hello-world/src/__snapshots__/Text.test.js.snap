// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Text extends _react2.default.Component {



  constructor(props) {
    super(props);this.




    removeGreeting = function () {
      this.props.removeGreeting(this.props.ten);
    };this.state = { name: props.ten };this.removeGreeting = this.removeGreeting.bind(this);}

  render() {
    return (
      _react2.default.createElement("div", { className: "Text", __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this }, "Hi, ",
        this.state.name, "!",
        _react2.default.createElement("br", { __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }),
        _react2.default.createElement("button", { onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this }, "Remove Me!")));


  }} renders and matches our snapshot 1`] = `
<div
  className="Text"
>
  Hi, 
  Person
  !
  <br />
  <button
    onClick={[Function]}
  >
    Remove Me!
  </button>
</div>
`;
